src/Workspaces/CSharp/Portable/Classification/Worker.cs

```diff
@@ -262,5 +265,27 @@ private void ClassifyDisabledText(SyntaxTrivia trivia, SyntaxTriviaList triviaLi
                 AddClassification(trivia, ClassificationTypeNames.ExcludedCode);
             }
         }
+
+        private void ClassifyLiteralAndSuffix(TextSpan tokenSpan, string literalClassification, int suffixLength)
+        {
+            var split = tokenSpan.End - suffixLength;
+            AddClassification(TextSpan.FromBounds(tokenSpan.Start, split), literalClassification);
+            AddClassification(TextSpan.FromBounds(split, tokenSpan.End), ClassificationTypeNames.Keyword);
+        }
+
+        private static bool TryGetNumericLiteralSuffix(string tokenText, out int suffixLength)
+        {
+            // Walk left until the first digit and count the number of characters on the way.
+            // The lexer ensures that the token only ends in valid suffixes; invalid suffix characters are part of a different token.
+            suffixLength = 0;
+            for (var i = tokenText.Length - 1; i >= 0 && !IsDigit(tokenText[i]); i--)
```

Comment:
❗ This will not correctly handle `0x8FUL`

Reply:
Of course. Should now be fixed: when the number is clearly an integer literal (0b or 0x prefix), D and F (and M, but that's not hexadecimal) are now no longer valid suffix characters. That should now cover all cases for integer and real literals in the C# spec, I think?

![image](https://github.com/dotnet/roslyn/assets/823398/fb184a38-5bf7-44e1-9169-f3060d982815)